src/server/api/routers/searchLocation.ts

```diff
@@ -0,0 +1,51 @@
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "../trpc";
+import { Client } from "@googlemaps/google-maps-services-js";
+const client = new Client({});
+const GOOGLE_MAPS_API_KEY = process.env.GOOGLE_MAPS_API_KEY || "";
+if (!GOOGLE_MAPS_API_KEY) {
+    console.log("No gmaps API key, please set it")
+}
+export const searchRouter = createTRPCRouter({
+    getLocation: publicProcedure
+        .input(z.object({ locationQuery: z.string() }))
+        .query(async ({ input }) => {
+            try {
+                const response = await client.textSearch({
+                    params: {
+                        query: input.locationQuery,
+                        key: GOOGLE_MAPS_API_KEY
+                    },
+                    timeout: 1000,
+                });
+                const results = response.data.results;
+
+                if (results.length === 0) {
+                    return { error: "No results found" };
+                } else if (results.length === 1) {
+                    const place = results[0];
+                    return {
+                        location: place!.formatted_address,
+                        latitude: place!.geometry!.location.lat,
+                        longitude: place!.geometry!.location.lng,
+                        placeId: place!.place_id,
+                        name: place!.name,
+                    };
+                } else {
+                    return {
```

Comment:
we should still return a single-item list if one result only, then let the user confirm his choice in the FE